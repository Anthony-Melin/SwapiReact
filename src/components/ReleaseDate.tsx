import LabelValueFactory from "./__components__/LabelValueFactory.tsx";

export default LabelValueFactory('releasedate', 'Release date')
