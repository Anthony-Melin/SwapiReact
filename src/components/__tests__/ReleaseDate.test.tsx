import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import ReleaseDate from '../ReleaseDate.tsx'

describe('ReleaseDate', () => {
    beforeEach(() => {
        render(<ReleaseDate value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('releasedate')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('releasedate')).toHaveTextContent(
            'Release date: VALUE'
        )
    })
})
