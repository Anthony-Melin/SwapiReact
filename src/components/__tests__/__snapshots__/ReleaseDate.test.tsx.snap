// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReleaseDate > should match snapshot 1`] = `
<p
  data-testid="releasedate"
>
  Release date
  : 
  VALUE
</p>
`;
